index.js

```diff
@@ -1,18 +1,78 @@
+//Guessing Game Project - Burlington Code Academy
+// started 2.5.20
+// due Monday, Feb. 10 by 9 a.m.
+// Written by Rob Mitchell
+// Binary search algorithm version 
+
+//set the standard input 
 const readline = require('readline');
-const rl = readline.createInterface(process.stdin, process.stdout);
+const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+
+// starts game using function
+function guessGame() {
+  let guessCount = 0; //Sets initial count for guesses
+  let maxRange = 100; //Sets initial max range for guessing
+  let minRange = 0; //Sets initial minimum range for guessing
+  //sets up user
+  console.log("Let's play a guessing game where you (human) \nmake up a number and I (computer) try to guess it!")
+  //starts async function
+  async function start() {
+    let rightGuess = null;
+    //asks for an upper bound to the guessing range
+    maxRange = (await ask("To play I need a range of numbers to guess. How about \nbetween zero and your choice. How high can I guess?"))
+    //function that calculates the max number of guesses using binary search algorithm
+    //then console logs that for user
+    guessNum(maxRange);
+    //starts while loop to run while condition is no correct guess
+    while (rightGuess !== "y" || "yes") {
+      //to use binary search method, first guess will be median
+      //Then the guess re-sets to the new median between min and max range, rounding up
+      let numGuess = Math.round(maxRange - ((maxRange - minRange) / 2)); 
+      //starts guessing here 
+      rightGuess = (await ask("Is it... " + numGuess + "? (Y/N)?")).trim().toLowerCase();
+      //adds one to the guess count, in order to show number of guesses upon success
+      guessCount += 1;
+      // response when (if) computer guesses correctly - yes, a little attitude
+      if (rightGuess === "y") {
```

Comment:
Here, you may want to include a check before this if statement that determines if a user entered valid input. Something like:
```js
let validResponses = ['y', 'yes', 'yup']
while ( !validResponses.includes(rightGuess) ) {
  rightGuess = await ask('That is not a valid response, try again. ')
}
```
or you could use a check similar to what is in the while loop above:
```js
if (rightGuess === 'y' || rightGuess === 'yes') {
   // console.log victory message
}
```
otherwise, if the user types in yes, it will continue to the else branch which takes an action as if the user entered 'n' or 'no'